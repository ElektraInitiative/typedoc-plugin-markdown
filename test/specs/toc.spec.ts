import * as Handlebars from 'handlebars';
import { ProjectReflection, Reflection } from 'typedoc';

describe(`TOC:`, () => {
  let moduleReflection: Reflection;
  let classReflection: Reflection;

  describe(`(default)`, () => {
    let project: ProjectReflection;
    beforeAll(async () => {
      project = await global.bootstrap(['toc.ts']);
      moduleReflection = project;
      classReflection = project.getChildByName('SomeClass') as any;
    });

    test(`should display toc for module'`, () => {
      expect(
        global.compileHelper(Handlebars.helpers.toc, moduleReflection),
      ).toMatchSnapshot();
    });

    test(`should display toc for class'`, () => {
      expect(
        global.compileHelper(Handlebars.helpers.toc, classReflection),
      ).toMatchSnapshot();
    });
  });
  describe(`(hideInPageToc)`, () => {
    let project: ProjectReflection;
    beforeAll(async () => {
      project = await global.bootstrap(['toc.ts'], { hideInPageTOC: true });
      moduleReflection = (project as any).children[0];
      classReflection = project.getChildByName('SomeClass') as any;
    });

    test(`should not display toc for class'`, () => {
      expect(Handlebars.helpers.toc.call(classReflection)).toBeNull();
    });
  });
});
