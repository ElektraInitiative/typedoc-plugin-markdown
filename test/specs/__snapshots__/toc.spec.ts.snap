// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TOC: (default) should display toc for class' 1`] = `
"## Table of contents

### Constructors

- [constructor](SomeClass.md#constructor)

### Methods

- [someMethod](SomeClass.md#somemethod)
"
`;

exports[`TOC: (default) should display toc for module' 1`] = `
"## Table of contents

### Classes

- [SomeClass](SomeClass.md)

### Type Aliases

- [Answer](../modules.md#answer)
- [answer](../modules.md#answer-1)

### Variables

- [\\_answer](../modules.md#_answer)
- [answer](../modules.md#answer-2)
"
`;
